### PR TITLE
Nf ui auth downloaders

### PR DESCRIPTION
**Pull request by jgors**
first scenario of many to be filled out using httpretty to mock authentication via post & get requests to download data. test will need to be refractored to be generalized for other test cases that are outline in the inline comments.

### CODE REVIEW

**Comment by yarikoptic** [comment]
Can't check in the phone but I ask afraid that tttpretty might not be py3 ready yet... D'oh... Skip those tests for py3 for now then

**Comment by yarikoptic** [comment]
Before refactoring you might like do one more but cut paste, so you could actually see what to refactor

File: datalad/downloaders/tests/test_http.py [hunk 1]
```diff
@@ -123,26 +126,46 @@ def enter_new(self):
         # pop last used credential, so we would use "new" ones
         del self._fixed_credentials[0]
 
+
```
Comment (yarikoptic) [inline datalad/downloaders/tests/test_http.py:130]: we need to have smth like

```
@skip_if(not httpretty)
```

File: datalad/downloaders/tests/test_http.py [hunk 2]
```diff
@@ -15,11 +15,13 @@
 from ..http import DownloadError
 from ..http import HTMLFormAuthenticator
 from ..providers import Providers, Credential  # to test against crcns
+from ...support.cookies import CookiesDB
 
 # BTW -- mock_open is not in mock on wheezy (Debian 7.x)
 import httpretty
```
Comment (yarikoptic) [inline datalad/downloaders/tests/test_http.py]:
```
try:
   import httpretty
except ImportError:
  httpretty = None
```

File: datalad/downloaders/tests/test_http.py (hunk 1 above)
Comment (jgors) [inline datalad/downloaders/tests/test_http.py:130]: did you push those changes made this morning so i can grab that `skip_if` decorator?

Comment (yarikoptic) [inline datalad/downloaders/tests/test_http.py:130]: yes

Comment (jgors) [inline datalad/downloaders/tests/test_http.py:130]: to this same branch (`nf-ui-auth-downloaders`)?  i don't see it (and it says your last commit to that branch was 6 days ago on your fork (https://github.com/yarikoptic/datalad/tree/nf-ui-auth-downloaders)

Comment (yarikoptic) [inline datalad/downloaders/tests/test_http.py:130]: no -- it is already in its mother branch into which I have merged that
nf-ui-auth-downloaders... just fire up gitk --all and I bet you will see
it

## search for skip  if you don't

Yaroslav O. Halchenko
Center for Open Neuroscience     http://centerforopenneuroscience.org
Dartmouth College, 419 Moore Hall, Hinman Box 6207, Hanover, NH 03755
Phone: +1 (603) 646-9834                       Fax: +1 (603) 646-1419
WWW:   http://www.linkedin.com/in/yarik

Comment (jgors) [inline datalad/downloaders/tests/test_http.py:130]: when i `git grep skip_if` it's not there.  there are those other `skip_if_something` functions, but not the simple one you wrote this morning -- the plain `skip_if`.  what am i missing here?...and i'm not quite sure what you mean by mother branch, please clarify.

Comment (jgors) [inline datalad/downloaders/tests/test_http.py:130]: i found it.  you pushed that to your `rf-crawler-new-design` branch.

File: datalad/downloaders/tests/test_http.py [hunk 3]
```diff
@@ -9,26 +9,29 @@
...
-from ...tests.utils import serve_path_via_http, with_tree
-from ...tests.utils import swallow_logs
-from ...tests.utils import with_tempfile
-from ...tests.utils import use_cassette
-from ...tests.utils import SkipTest
+from ...tests.utils import assert_in, assert_not_in, assert_equal, assert_false, \
+                assert_raises, ok_file_has_content, serve_path_via_http, \
+                with_tree, swallow_logs, with_tempfile, use_cassette, \
+                SkipTest, skip_if, assert_greater
+
```
Comment (yarikoptic) [inline datalad/downloaders/tests/test_http.py:34]: why oh why?

**Comment by yarikoptic** [comment]
merged locally and fixed it up after

**Pull request closed without merging**